libr/asm/p/asm_xcore_cs.c

```diff
@@ -1,60 +0,0 @@
-/* radare2 - LGPL - Copyright 2014-2021 - pancake */
-
-#include <r_asm.h>
-#include <r_lib.h>
-#include "cs_version.h"
-
-static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
-	csh handle;
-	cs_insn* insn;
-	int mode, n, ret = -1;
-	mode = a->config->big_endian? CS_MODE_BIG_ENDIAN: CS_MODE_LITTLE_ENDIAN;
-	memset (op, 0, sizeof (RAsmOp));
-	op->size = 4;
-	ret = cs_open (CS_ARCH_XCORE, mode, &handle);
-	if (ret) {
-		goto fin;
-	}
-	cs_option (handle, CS_OPT_DETAIL, CS_OPT_OFF);
-	n = cs_disasm (handle, (ut8*)buf, len, a->pc, 1, &insn);
-	if (n < 1) {
-		r_asm_op_set_asm (op, "invalid");
-		op->size = 4;
-		ret = -1;
-		goto beach;
-	}
-	ret = 4;
-	if (insn->size < 1) {
-		goto beach;
-	}
-	op->size = insn->size;
-	r_strf_buffer (256);
-	r_asm_op_set_asm (op, r_strf ("%s%s%s",
-		insn->mnemonic, insn->op_str[0]? " ": "",
-		insn->op_str));
-	// TODO: remove the '$'<registername> in the string
```

Comment:
We might want to open a GitHub issue for this one.

Reply:
there are many TODO messages already, imho its better than holding tons of tickets for these small things if nobody complains or feels the need to have X or Y but yeah if you remove the TODO. open a ticket so we dont forget